engine/src/main/java/org/pentaho/di/job/entries/filesexist/JobEntryFilesExist.java

```diff
@@ -0,0 +1,264 @@
+    }
+    retval.append( "      " ).append( XMLHandler.addTagValue( "filename", filename ) );
+
+    retval.append( "      <fields>" ).append( Const.CR );
+    if ( arguments != null ) {
+      for ( int i = 0; i < arguments.length; i++ ) {
+        retval.append( "        <field>" ).append( Const.CR );
+        retval.append( "          " ).append( XMLHandler.addTagValue( "name", arguments[i] ) );
+        retval.append( "        </field>" ).append( Const.CR );
+        if ( parentJobMeta != null ) {
+          parentJobMeta.getNamedClusterEmbedManager().registerUrl( arguments[i] );
+        }
+      }
+    }
+    retval.append( "      </fields>" ).append( Const.CR );
+
+    return retval.toString();
+  }
+
+  public void loadXML( Node entrynode, List<DatabaseMeta> databases, List<SlaveServer> slaveServers,
+    Repository rep, IMetaStore metaStore ) throws KettleXMLException {
+    try {
+      super.loadXML( entrynode, databases, slaveServers );
+      filename = XMLHandler.getTagValue( entrynode, "filename" );
+
+      Node fields = XMLHandler.getSubNode( entrynode, "fields" );
+
+      // How many field arguments?
+      int nrFields = XMLHandler.countNodes( fields, "field" );
+      allocate( nrFields );
+
+      // Read them all...
+      for ( int i = 0; i < nrFields; i++ ) {
+        Node fnode = XMLHandler.getSubNodeByNr( fields, "field", i );
+
+        arguments[i] = XMLHandler.getTagValue( fnode, "name" );
+
+      }
+    } catch ( KettleXMLException xe ) {
+      throw new KettleXMLException( BaseMessages.getString(
+        PKG, "JobEntryFilesExist.ERROR_0001_Cannot_Load_Job_Entry_From_Xml_Node", xe.getMessage() ) );
+    }
+  }
+
+  public void loadRep( Repository rep, IMetaStore metaStore, ObjectId id_jobentry, List<DatabaseMeta> databases,
+    List<SlaveServer> slaveServers ) throws KettleException {
+    try {
+      filename = rep.getJobEntryAttributeString( id_jobentry, "filename" );
+
+      // How many arguments?
+      int argnr = rep.countNrJobEntryAttributes( id_jobentry, "name" );
+      allocate( argnr );
+
+      // Read them all...
+      for ( int a = 0; a < argnr; a++ ) {
+        arguments[a] = rep.getJobEntryAttributeString( id_jobentry, a, "name" );
+      }
+    } catch ( KettleException dbe ) {
+      throw new KettleException( BaseMessages
+        .getString( PKG, "JobEntryFilesExist.ERROR_0002_Cannot_Load_Job_From_Repository", "" + id_jobentry, dbe
+          .getMessage() ) );
+    }
+  }
+
+  public void saveRep( Repository rep, IMetaStore metaStore, ObjectId id_job ) throws KettleException {
+    try {
+      rep.saveJobEntryAttribute( id_job, getObjectId(), "filename", filename );
+
+      // save the arguments...
+      if ( arguments != null ) {
+        for ( int i = 0; i < arguments.length; i++ ) {
+          rep.saveJobEntryAttribute( id_job, getObjectId(), i, "name", arguments[i] );
+        }
+      }
+    } catch ( KettleDatabaseException dbe ) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: While wrapping the caught exception into a custom one, information about the caught exception is being lost, including information about the stack trace of the exception.

**Fix**: If the caught exception object does not contain sensitive information, consider passing it as the &#34;rootCause&#34; or inner exception parameter to the constructor of the new exception before throwing the new exception. (Note that not all exception constructors support inner exceptions. Use a wrapper exception that supports inner exceptions.) 
[Learn more](https://www.ibm.com/support/pages/best-practice-catching-and-re-throwing-java-exceptions)